### ISSUE TITLE
summary

### ISSUE THREAD

**Issue by Xiaodiudiudiu** [Xiaodiudiudiu/summary#1]
### 项目问题：

我在安装了requirement.txt包之后，许多库发生了冲突 （inconsistency）   新建一个env 代替 base

Q 需要把两个project 分开吗？

<img src="C:\Users\zzyhx\AppData\Roaming\Typora\typora-user-images\image-20220114155348354.png" alt="image-20220114155348354" style="zoom:78%;" />

延申：遇到很多inconsistency怎么办呢

### Q2

tensorflow & keras 

### 

#### model:

1. 将data (labeled)分为 training & test data 
2. split input & output
3. define model (设置layer & 激活函数)
4. compile model ( loss, **optimizer**, metric)
5. fit 
6. evaluation
7. inference or prediction

##### 建立 word embedding

sentence -> each word assigned int

i.e.

given volcabulary size  

```
the glass of milk
```

```
[1526, 380, 9012, 4197] onehot? 有一本自带的百科全书，标注了每个词的
```

<img src="C:\Users\zzyhx\AppData\Roaming\Typora\typora-user-images\image-20220114163400025.png" alt="image-20220114163400025" style="zoom:50%;" />

因为每句话初始长度不一样， set a fixed length **(padding)**

### dim = 10。 用10个feature 表示一个 word？（features 怎么自己定义呢？ 看起来是encoder自动生成了 given fixed length vector）

```
[-0.01312741 -0.00770368  0.01743542 -0.04066963 -0.0198542
    0.02775803 -0.04056864  0.01815999  0.01143856 -0.00908693] initializing?
```

![image-20220114163046016](C:\Users\zzyhx\AppData\Roaming\Typora\typora-user-images\image-20220114163046016.png)

#### 最小语义解释

define documents

docs = ['Well done!',
		'Good work',
		'Great effort',
		'nice work',
		'Excellent!',
		'Weak',
		'Poor effort!',
		'not good',
		'poor work',
		'Could have done better.']

define class labels

labels = array([1,1,1,1,1,0,0,0,0,0])

#### 对语义的情绪做判断（情绪分析）

```
[[37, 77], [47, 1], [47, 82], [62, 1], [86], [17], [40, 82], [32, 47], [40, 1], [64, 14, 77, 18]]
```

暂时只是发现，只有出现过的单词，在训练次数足够高的时候，accuracy才足够高，是否需要meta

且单词的顺序也会影响情绪的判断

skip-gram CBOW 用哪一个呢？ 都用

<img src="C:\Users\zzyhx\AppData\Roaming\Typora\typora-user-images\image-20220114164333893.png" alt="image-20220114164333893" style="zoom:50%;" />

i-2 i-1 i  i+1 i+2

<img src="C:\Users\zzyhx\AppData\Roaming\Typora\typora-user-images\image-20220114164442475.png" alt="image-20220114164442475" style="zoom:50%;" />

cat embedding  -  类似于上文的10 features?

softmax get inference

training test

word2vec 会常用

**corpus** = ['king is a strong man', 
          'queen is a wise woman', 
          'boy is a young man',
          'girl is a young woman',
          'prince is a young king',
          'princess is a young queen',
          'man is strong', 
          'woman is pretty',
          'prince is a boy will be king',
          'princess is a girl will be queen']

### Q

**去掉stop words** - 中文对应的是什么呢？或者说 不是 stop words 的是 term 与 态度转折词 吗

i.e. 

他经常**生气** - ，**但  +   愉快**的时候**不**会生气

句间逻辑可以分析吗？ 可以，去找有否定意义连接词

段间逻辑可以分析吗？

```
['king strong man',
 'queen wise woman',
 'boy young man',
 'girl young woman',
 'prince young king',
 'princess young queen',
 'man strong',
 'woman pretty',
 'prince boy king',
 'princess girl queen']
```

seq2word

```
{'boy',
 'girl',
 'king',
 'man',
 'pretty',
 'prince',
 'princess',
 'queen',
 'strong',
 'wise',
 'woman',
 'young'}
 
```

CBOW  context words are averaged before predicting the center word?

```
{'king': 0, 'prince': 1, 'wise': 2, 'queen': 3, 'woman': 4, 'princess': 5, 'strong': 6, 'boy': 7, 'girl': 8, 'pretty': 9, 'man': 10, 'young': 11}
```

**这里没有用word size + one hot encoder ，而是依次key - value 赋值的**

把sentence 里的每一个word split up

```
[['king', 'strong', 'man'], ['queen', 'wise', 'woman'], ['boy', 'young', 'man'], ['girl', 'young', 'woman'], ['prince', 'young', 'king'], ['princess', 'young', 'queen'], ['man', 'strong'], ['woman', 'pretty'], ['prince', 'boy', 'king'], ['princess', 'girl', 'queen']]

?相当于 059, 324 （体现哪些词会一起出现）

感觉分析的 都是  句内逻辑！


window_size = 2 

[['king', 'strong'], ['king', 'man'], ['strong', 'king'], ['strong', 'man'], ['man', 'king'], ['man', 'strong'], ['queen', 'wise'], ['queen', 'woman'], ['wise', 'queen'], ['wise', 'woman'], ['woman', 'queen'], ['woman', 'wise'], ['boy', 'young'], ['boy', 'man'], ['young', 'boy'], ['young', 'man'], ['man', 'boy'], ['man', 'young'], ['girl', 'young'], ['girl', 'woman'], ['young', 'girl'], ['young', 'woman'], ['woman', 'girl'], ['woman', 'young'], ['prince', 'young'], ['prince', 'king'], ['young', 'prince'], ['young', 'king'], ['king', 'prince'], ['king', 'young'], ['princess', 'young'], ['princess', 'queen'], ['young', 'princess'], ['young', 'queen'], ['queen', 'princess'], ['queen', 'young'], ['man', 'strong'], ['strong', 'man'], ['woman', 'pretty'], ['pretty', 'woman'], ['prince', 'boy'], ['prince', 'king'], ['boy', 'prince'], ['boy', 'king'], ['king', 'prince'], ['king', 'boy'], ['princess', 'girl'], ['princess', 'queen'], ['girl', 'princess'], ['girl', 'queen'], ['queen', 'princess'], ['queen', 'girl']]
```

**可以发现，这个例子中，每一句话各个单词都是顺逻辑，没有转，所有聚类实现了相近意思的单词聚集**

怎么实现聚类的呢？

把前者设为input 后者设为label

<img src="C:\Users\zzyhx\AppData\Roaming\Typora\typora-user-images\image-20220114170633347.png" alt="image-20220114170633347" style="zoom:67%;" />

convert them to int   5 - 10, 10 - 5, 1 - 0, 0 - 1

```
{'woman': 0,
 'pretty': 1,
 'wise': 2,
 'queen': 3,
 'king': 4,
 'man': 5,
 'young': 6,
 'prince': 7,
 'princess': 8,
 'girl': 9,
 'strong': 10,
 'boy': 11}
```

convert them to numpy arrays

<img src="C:\Users\zzyhx\AppData\Roaming\Typora\typora-user-images\image-20220114183927391.png" alt="image-20220114183927391" style="zoom:67%;" />

3d?

bert + pre-train 看文献 + 实操 + simbert模型

<img src="C:\Users\zzyhx\AppData\Roaming\Typora\typora-user-images\image-20220114163305224.png" alt="image-20220114163305224" style="zoom:50%;" />

### Q4 

远程服务器下包这类的问题 root

#### Q5

我需要把下的包放到我的env里面吗，服务器的linux 文件夹 也可以当环境吗？

或者，虚拟环境只能放到env里面吗？ yes 

一个全新的词可以用average

### Pytorch1.7 cuda 10.1

ctrl +R 可以 集体确定

txt -> json 

ctrl + shift + E 

分段的 console执行语句

模型架构 transformer + LSTM + c什么什么的
